### PR TITLE
Added javscript questions

### PR DESCRIPTION
**Pull request by DarshanVaishya**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

 #14

### CODE REVIEW

File: src/data/full-quiz.js [hunk 1]
```diff
@@ -1,4 +1,76 @@
...
+    Distractor3: "None of given",
+    Explanation: "The bind functions returns a new function with given arguement as the new function's this keyword.",
+    Link: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind",
+  },
+  {
+    Question: "What is a first class function?",
+    Answer: "The ability to send and receive functions like value.",
+    Distractor1: "Functions with higher priority than others.",
+    Distractor2: "Functions declared in classes",
+    Distractor3: "None of given",
```
Comment (jdwilkin4) [inline src/data/full-quiz.js:70]: None of given doesn't sound quite right to me. Maybe we can alter that answer. 
Proposed fix:
Distractor3: "Functions that only return numbers",

File: src/data/full-quiz.js [hunk 2]
```diff
@@ -1,4 +1,76 @@
...
+    Explanation: "If functions can be sent and received just like values, than that languge is said to have first class functions. They are also called first class citizens.",
+    Link: "https://www.geeksforgeeks.org/difference-between-first-class-and-higher-order-functions-in-javascript/",
```
Comment (jdwilkin4) [inline src/data/full-quiz.js:72]: For the link field we only want to use freeCodeCamp articles or official docs.

Proposed fix:
Link:"https://www.freecodecamp.org/news/discover-the-power-of-first-class-functions-fd0d7b599b69/"

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Hi @DarshanVaishya ! 

Thank you so much for your questions. 

Please read through my comments to make those changes before we merge it into the site. 👍

**Comment by DarshanVaishya** [comment]
I have implemented these proposed fixes. Apologies for minor mistakes such as capitalization.

File: src/data/full-quiz.js [hunk 3]
```diff
@@ -5,71 +5,71 @@ const fullQuiz = [
...
     Distractor1: "let",
     Distractor2: "const",
     Distractor3: "class",
-    Explanation: "var is hoisted with default value of undefined. While let, const and classes are hoisted but are in Temporal Dead Zone(TDZ) until declaration is executed.",
+    Explanation: "var is hoisted with the default value of undefined while let, const and classes are hoisted but are in the Temporal Dead Zone(TDZ) until the declaration is executed.",
     Link: "https://developer.mozilla.org/en-US/docs/Glossary/Hoisting",
   },
   {
-    Question: "Which of the following expression will return true in Javascript?",
+    Question: "Which of the following expression will return true in JavaScript?",
```
Comment (jdwilkin4) [inline src/data/full-quiz.js:39]: expression should be expressions.
Which of the following expressions will return true in JavaScript?

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for making those changes. 
If you could make that one change mentioned below, we can merge your PR into the main branch.  👍

File: src/data/full-quiz.js (hunk 3 above)
Comment (DarshanVaishya) [reply src/data/full-quiz.js]: Done!

**Review by jdwilkin4 — APPROVED** [review]
LGTM! 👍

**Comment by jdwilkin4** [comment]
Thank you for your quiz questions! 
We look forward to future contributions 👍

**Pull request merged**